proto/config/config.proto

```diff
@@ -0,0 +1,89 @@
+syntax = "proto3";
+
+package linkerd2.config;
+
+option go_package = "github.com/linkerd/linkerd2/controller/gen/config";
+
+// ConfigStore provides an abstraction over the underlying configuration storge.
+service ConfigStore {
+  // Return the JSON-serialized global config.
+  rpc GetGlobalConfig(GetParam) returns (GlobalConfig) {}
+
+  // Return the JSON-serialized proxy config.
+  rpc GetProxyConfig(GetParam) returns (ProxyConfig) {}
+
+  // Set the global config. It overwrites the existing value.
+  rpc SetGlobalConfig(GlobalConfig) returns (Result) {}
+
+  // Set the proxy config. It overwrites the existing value.
+  rpc SetProxyConfig(ProxyConfig) returns (Result) {}
+}
+
+message GlobalConfig {
+  string linkerd_namespace = 1;
+  bool cni_enabled = 2;
+  bool tls_enabled = 3;
```

Comment:
Not as a boolean ;)

We'll need to store things like the identity trust anchors, though...

I'd say that a boolean is fine for now, if we need to support the existing config surface area, but it will _definitely_ be deprecated in the near future (and so you may opt to use a high-numbered field ID).

Reply:
Shall we replace the TLS enum (defined below) with a `message` so that we can introduce additional properties like identity trust anchors etc. into it, without affecting the `GlobalConfig` at this level?